src/ipvs/ip_vs_proto_tcp.c

```diff
@@ -636,7 +636,7 @@ static int tcp_fnat_in_handler(struct dp_vs_proto *proto,
         dev = conn->in_dev;
 
     if (likely(dev && (dev->flag & NETIF_PORT_FLAG_TX_TCP_CSUM_OFFLOAD))) {
-        if (af == AF_INET6) {
+        if (AF_INET6 == af) {
             mbuf->l4_len = ntohs(ip6_hdr(mbuf)->ip6_plen);
             mbuf->l3_len = iphdrlen;
```

Comment:
`l3_len` should  be NOT include exthdr lenght for IPv6 upper layer checksum.
See `https://tools.ietf.org/html/rfc2460#page-27` for detail.